src/OrleansRuntime/Catalog/DefaultGrainActivator.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Globalization;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// <see cref="IGrainActivator"/> that uses type activation to create grains.
+    /// </summary>
+    public class DefaultGrainActivator : IGrainActivator
+    {
+        private readonly Func<Type, ObjectFactory> createFactory = type => ActivatorUtilities.CreateFactory(type, Type.EmptyTypes);
```

Comment:
Unfortunately not, since the need is to also perform some cleanup tasks after the instance is done with. Also, this was part of the work started in #1901 where we wanted to emulate MVC's controller activation strategy, which to be honest, make a lot of sense for us too.
This matches 1 to 1 with [DefaultControllerActivator](https://github.com/aspnet/Mvc/blob/760c8f38678118734399c58c2dac981ea6e47046/src/Microsoft.AspNetCore.Mvc.Core/Controllers/DefaultControllerActivator.cs) (except that they have an extra layer of abstraction for ITypeActivatorCache which I didn't add here, as it doesn't add any value yet)